propagator/vitess/lib/opentelemetry/propagator/vitess/sql_query_propagator.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# OpenTelemetry is an open source observability framework, providing a
+# general-purpose API, SDK, and related tools required for the instrumentation
+# of cloud-native software, frameworks, and libraries.
+#
+# The OpenTelemetry module provides global accessors for telemetry objects.
+# See the documentation for the `opentelemetry-api` gem for details.
+module OpenTelemetry
+  # Namespace for OpenTelemetry propagator extension libraries
+  module Propagator
+    # Namespace for OpenTelemetry Vitess propagation
+    module Vitess
+      # Setter for Vitess SQL query propagation
+      module SqlQuerySetter
+        extend self
+
+        # Set a key and value on the carrier. Assumes the carrier is a string.
+        # The key and value will be wrapped in a comment block prepended to the carrier.
+        #
+        # @param [String] carrier The carrier to set the key and value on
+        # @param [String] key The key to set
+        # @param [String] value The value to set
+        def set(carrier, key, value)
+          carrier.gsub!(/\A/, "/*#{key}=#{value}*/")
+        end
+      end
+
+      # Propagates context using Vitess header format:
+      # https://vitess.io/docs/16.0/user-guides/configuration-advanced/tracing/#instrumenting-queries
+      class SqlQueryPropagator
+        VT_SPAN_CONTEXT = 'VT_SPAN_CONTEXT'.freeze
+        FIELDS = [VT_SPAN_CONTEXT].freeze
+
+        private_constant :VT_SPAN_CONTEXT, :FIELDS
+
+        def initialize
+          @jaeger = OpenTelemetry::Propagator::Jaeger.text_map_propagator
+        end
+
+        # No-op extractor.
+        def extract(carrier, context: Context.current, getter: nil)
+          context
+        end
+
+        # @param [Object] carrier to update with context.
```

Comment:
We'll always expect a string here, right?
```suggestion
        # @param [String] carrier to update with context.
```

Reply:
No. The spec says it can be anything, and the setter needs to be compatible with the carrier. I added a test (pushing in a sec) to show this.